plugins/MultiSites/Dashboard.php

```diff
@@ -73,20 +80,20 @@ public function __construct($period, $date, $segment)
 
     public function setSitesTable(DataTable $sites)
     {
-        $this->numSites     = $sites->getRowsCount();
         $this->sitesByGroup = $this->moveSitesHavingAGroupIntoSubtables($sites);
+        $this->rememberNumberOfSites();
     }
 
     public function getSites($request, $limit)
     {
-        $request['filter_limit'] = $limit;
+        $request['filter_limit']  = $limit;
+        $request['filter_offset'] = isset($request['filter_offset']) ? $request['filter_offset'] : 0;
 
-        $sitesExpanded = $this->convertDataTableToArrayAndApplyFilters($this->sitesByGroup, $request);
-        $sitesFlat     = $this->makeSitesFlat($sitesExpanded);
-        $sitesFlat     = $this->applyLimitIfNeeded($sitesFlat, $limit);
-        $sitesFlat     = $this->enrichValues($sitesFlat);
+        $this->makeSitesFlatAndApplyGenericFilters($this->sitesByGroup, $request);
```

Comment:
We do no longer make the sites flat after all filters were run, instead we need to make the sites array flat and then apply all the other filters.